src/toast/util/toastReducer.ts

```diff
@@ -57,6 +64,29 @@ function toastReducer(state: ToastState, action: ToastAction): ToastState {
       break;
     }
 
+    case "SET_LIMIT": {
+      const {limit} = action;
+      // prune the toasts exceeding the limit
+      const toastStack = state.toastStack.slice(
+        Math.max(0, state.toastStack.length - limit)
+      );
+
+      newState = {
+        ...state,
+        toastStack,
+        limit
+      };
+      break;
+    }
+    case "SET_AUTO_CLOSE": {
+      const {autoCloseToasts} = action;
+
+      newState = {
+        ...state,
+        autoCloseToasts
```

Comment:
```suggestion
      newState = {
        ...state,
        autoCloseToasts: action.autoCloseToasts
```
It is not used anywhere else, so we can directly use it inside